pkgs/applications/networking/browsers/palemoon/default.nix

```diff
@@ -1,29 +1,57 @@
-{ stdenv, lib, fetchFromGitHub, writeScript, desktop-file-utils
-, pkg-config, autoconf213, alsaLib, bzip2, cairo
-, dbus, dbus-glib, ffmpeg, file, fontconfig, freetype
-, gnome2, gnum4, gtk2, hunspell, libevent, libjpeg
-, libnotify, libstartup_notification, wrapGAppsHook
-, libGLU, libGL, perl, python2, libpulseaudio
-, unzip, xorg, wget, which, yasm, zip, zlib
-
-, withGTK3 ? true, gtk3
+# https://developer.palemoon.org/build/linux/
```

Comment:
Why this commentary at the top?

Reply:
It's from the stdenv override in all-packages.nix that I had to remove to fix WebGL on my machine, I moved it here because I didn't want to lose the link completely. The compiler guideline there is still relevant for the official branding we use.

> GCC versions 4.9, 5.x, 7.x, 8.x, 9.x or 10.x
> 
>   - GCC 6.x is not _at all_ recommended and has been known to produce unstable binaries. It has been reported that Debian-based GCC 6.x seems to produce stable enough binaries. Your mileage will vary, of course.
>   - GCC 9.x and 10.x are still considered experimental at this time, however, they have thus far produced stable binaries.

I can add more context to the comment if you want. I had also planned to `assert` if a supported stdenv is used so a future GCC bump in stdenv wouldn't pull palemoon into an unsupported setup. Or does that seem too "overkill" to you?

Reply:
Added an explanation for now.

Reply:
Hum... The explanation is fine for now. Setting GCC8 in all-packages.nix is a better way to handle this, and the assert would be useful only to block GCC6 and older versions, not the newer ones. After all, the tendency is that they use newer compilers, therefore banning newer compilers makes little sense.

Reply:
> Setting GCC8 in all-packages.nix is a better way to handle this

I agree, but that is what broke GL loading for me in the first place so I consider that not-viable.

> and the assert would be useful only to block GCC6 and older versions, not the newer ones. After all, the tendency is that they use newer compilers, therefore banning newer compilers makes little sense.

Their tendency is actually to build releases on a CentOS 7 system with GCC 7.x (or maybe 8.x in the future) until June 30th 2024, [last thing I remember](https://forum.palemoon.org/viewtopic.php?f=37&t=23031&start=20#p175838). I do see your point though.

Reply:
I see your points now. There is a coupling among OpenGL, Palemoon and GCC.